### ISSUE TITLE
Crashes occur when autocomplete suggestions appear while working with a DataFrame that includes column names in Japanese.

### ISSUE THREAD

**Issue by ujtwr** [rstudio/rstudio#14625]
Verify fixed in:
- [ ] Patch: 2024.04.1
- [ ] Release: 2024.07.0

---

### System details

    RStudio Edition : Server
    RStudio Version : 2024.04.0 Build 735
    OS Version      :  Ubuntu 22
    R Version       :  4.4.0

### Steps to reproduce the problem
1. Create a DataFrame with Japanese column names
2. Creating a plot using ggplot
3. Crashes when autocomplete suggestions appear. Attached is the screen shot immediately after the crash.

Pressing the tab key after "y =" or when autocomplete suggestions appear results in a state similar to the screenshot.

![image](https://github.com/rstudio/rstudio/assets/42717412/9d813f81-2ca4-4cba-bb5a-ad61eb528425)

The server logs are outputting as follows:

```
5月 01 14:50:56 rstudio rsession-*******[3918]: ERROR Unexpected exception: vector<bool>::_M_range_check: __n (which is 184467\44073709551587) >= this->size() (which is 65535); LOGGED FROM: int rstudio::r::session::RReadConsole(const char*, CONSOLE_BUFFER_CHAR*\, int, int) src/cpp/r/session/RStdCallbacks.cpp:428
5月 01 14:50:56 rstudio rsession-*******[3918]: ERROR R SUICIDE: Unexpected exception: vector<bool>::_M_range_check: __n (whic\h is 18446744073709551587) >= this->size() (which is 65535); LOGGED FROM: void {anonymous}::rSuicide(const string&) src/cpp/session/Se\ssionMain.cpp:1264
5月 01 14:51:04 rstudio rsession-******[4016]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core\::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:746
```

### Describe the problem in detail

Crashes occur.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@ujtwr Thank you for raising this! I can reproduce this in RStudio Server 2024.04.0+735 on Ubuntu 22 via Chrome on Windows 11. It works correctly in RStudio Desktop 2024.04.0+735 on Windows. I can also reproduce this with other languages, like Hebrew and Arabic. We'll review this bug.

To reproduce:

Japanese
```
library(tibble)
library(ggplot2)
d <-
  tibble(
    ちちち = rnorm(10),
    面面面 = rnorm(10)
  )

d %>%
  ggplot(mapping = aes(x = ちちち, y = ))
```

Hebrew & Arabic
```
library(tibble)
library(ggplot2)
d <-
  tibble(
    ללל= rnorm(10),
    ششش = rnorm(10)
  )

d %>%
  ggplot(mapping = aes(x = ללל, y = ))
```

**Comment by ronblum** [comment]
This is also occurring with R 4.3.3. Combination of RStudio Server and R:
- 2023.12.1+402 + 4.3.3 & 4.4.0: Works
- 2024.04.0+735 + 4.3.3 & 4.4.0: Crashes

**Comment by kevinushey** [comment]
Interestingly enough, I'm not able to reproduce with the ARM64 builds of RStudio on Ubuntu 22. I'll see if I can reproduce on x64...

**Comment by kevinushey** [comment]
Crash happens here:

https://github.com/rstudio/rstudio/blob/ce04a06f67c829fbcee68a67248b1f4df0d3926b/src/cpp/session/modules/SessionRCompletions.cpp#L181-L182

https://github.com/rstudio/rstudio/blob/ce04a06f67c829fbcee68a67248b1f4df0d3926b/src/cpp/core/StringUtils.cpp#L597-L607

**Comment by kevinushey** [comment]
Impressive that we've successfully managed to find a bug in 13 year old code!

**Comment by ronblum** [comment]
Verified fixed using examples above in RStudio Server 2024.04.1+748 and 2024.07.0-daily+102 on Ubuntu 22 via Chrome on Windows 11.

**Comment by benz0li** [comment]
Workaround: E.g. French & German

```R
d <-
  tibble(
    "allé" = rnorm(10),
    "bär" = rnorm(10)
  )

d %>%
  ggplot(mapping = aes(x = "bär", y = ))
```

**Comment by ronblum** [comment]
The patched release of RStudio Desktop, 2024.04.1+748, contains the fix is now live: https://posit.co/download/rstudio-desktop/

**Comment by ronblum** [comment]
QA note: Added to [autocomplete tests](https://github.com/rstudio/qa/blob/master/Products/RStudio_IDE/autocomplete/autocomplete-testing.qmd).
